akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -135,6 +135,7 @@ abstract class SerializerWithStringManifest extends Serializer {
    * and message is dropped. Other exceptions will tear down the TCP connection
    * because it can be an indication of corrupt bytes from the underlying transport.
    */
+  @throws(classOf[NotSerializableException])
```

Comment:
Java implementations of SerializerWithStringManifest allow to throw java.io.NotSerializableException now